ballerina/records.bal

```diff
@@ -269,3 +269,16 @@ type ParseResult record {|
     parser:DocumentNode document;
     ErrorDetail[] validationErrors;
 |};
+
+type Result record {|
+    readonly anydata key;
+    anydata|error value;
+|};
+
+type Key record {|
+    readonly anydata key;
+|};
+
+type PlaceHolderNode record {|
+    string __uuid;
+|};
```

Comment:
Kept these inside the graphql module instead of dataloader submodule, since we don't have to expose these types to the user. This types are only used internally by the graphql engine .